### ISSUE TITLE
Bookmarks search does not find duplicate entries

### ISSUE THREAD

**Issue by WebworkrNet** [mozilla-mobile/fenix#27508]
### Steps to reproduce

1. I manually discovered several bookmarks with identical title in advance.
2. Bookmark: Open search
3. I search for the bookmark with appropriate search term(s).

### Expected behaviour

I expect, if multiple entries are present, that this is also reflected in the result set - i.e. all are displayed, even if they are present multiple times.

### Actual behaviour

The bookmark is output only once as a search result. Either the search stops immediately after the first hit or results are combined into one search result from only one entry. But this way I can't search for bookmarks that may exist multiple times without further tools.

### Device name

Samsung Galaxy A13 5G (SM-A136B/DSN)

### Android version

Android 12

### Firefox release type

Firefox

### Firefox version

105.1.0 (Build #1051000), b1e7cbeeb+ AC: 105.0.7, e2aeb82d54 GV: 105.0-20220922015904 AS: 94.1.0

### Additional information

Fennec (F-Droid)

**Comment by SoftVision-LorandJanos** [comment]
Hi @WebworkrNet! Thanks for your report!
I couldn't reproduce your issue, if you give the same title to different bookmarked pages, I think it would be expected for more than one to be displayed when searching for it/them. And the same page cannot be bookmarked twice.
Could you show me an example in a video, maybe? Thank you!

**Comment by WebworkrNet** [comment]
> I couldn't reproduce your issue, if you give the same title to different bookmarked pages, I think it would be expected for more than one to be displayed when searching for it/them. 

> Could you show me an example in a video, maybe? Thank you!

Can do.
I will probably need some time for this.

**Comment by SoftVision-LorandJanos** [comment]
I managed to reproduce your issue using the steps provided to create duplicate bookmarks.
I also tried to have the identical name but slightly different URLs on different devices, where they were displayed separately in the search results.
But if I understand this correctly, I believe your actual result is expected.
What I understand is that if you have identical bookmarks from different devices, they should be displayed separately, but if they are identical, I believe it's expected to be displayed only once in bookmark search.
